The AMPL linear program that you type into the file parallels the algebraic form in every respect. It specifies the decision variables, defines the objective, and lists the constraints. It differs mainly in being somewhat more formal and regular, to facilitate computer processing. Each variable is named in a var statement, and each constraint by a statement that begins with subject to and a name like Time or B_limit for the constraint. Multiplication requires an explicit * operator, and the ≤ relation is written <=.

In [1]:
from amplpy import AMPL, Environment

In [3]:
#this is for windows only. after dockerizing not required anymore.
#ampl = AMPL(Environment('D:\\amplide.mswin64\\amplide.mswin64')) 
ampl = AMPL(Environment('/opt/ampl.linux64')) 

In [4]:
#ampl.option['solver'] = 'cplexamp'
#we can do this to set the solver we want.

In [5]:
ampl.read('./Model1.mod') #read the model
#reads the file into AMPL, just as if you had 
#typed it line-by-line at ampl: prompt in the ampl IDE

In [6]:
ampl.solve() 
#(OR): ampl.eval('objective Profit; solve;')
#After we read the model, all we need to do is solve().

MINOS 5.51: optimal solution found.
2 iterations, objective 270000

"option abs_boundtol 9.094947017729282e-13;"
or "option rel_boundtol 1.5158245029548803e-16;"
will change deduced dual values.



The message MINOS 5.5 directly following the solve command indicates that AMPL used version 5.5 of a solver called MINOS. We have used MINOS and several other solvers for the examples in this book. You may have a different collection of solvers available on your computer, but any solver should give you the same optimal objective value for a linear program. Often there is more than one solution that achieves the optimal objective, however, in which case different solvers may report different optimal values for the variables. (Commands for choosing and controlling solvers will be explained in Section 11.2.)